### PR TITLE
[feature](nereids) Support to get partition related table from mv and check the query operator

### PR DESCRIPTION
**Pull request by seawinde**
## Proposed changes

**Function 1**
check the select query plan is contain the stmt as following or not

>      SELECT
>      [hint_statement, ...]
>      [ALL | DISTINCT | DISTINCTROW | ALL EXCEPT ( col_name1 [, col_name2, col_name3, ...] )]
>      elect_expr [, select_expr ...]
>      [FROM table_references
>      PARTITION partition_list]
>      [TABLET tabletid_list]
>      [TABLESAMPLE sample_value [ROWS | PERCENT]
>      [REPEATABLE pos_seek]]
>      [WHERE where_condition]
>      [GROUP BY [GROUPING SETS | ROLLUP | CUBE] {col_name | expr | position}]
>      [HAVING where_condition]
>      [ORDER BY {col_name | expr | position}
>      [ASC | DESC], ...]
>      [LIMIT {[offset,] row_count | row_count OFFSET offset}]
>      [INTO OUTFILE 'file_name']

>      if analyzedPlan contains the stmt as following:
>      PARTITION partition_list]
>      [TABLET tabletid_list] or
>      [TABLESAMPLE sample_value [ROWS | PERCENT]
>              [REPEATABLE pos_seek]]

     this method will return true.

**Function 2**

> Get related base table info which materialized view plan column reference,
> input param plan should be rewritten plan that sub query should be eliminated

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by seawinde** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java [hunk 1]
```diff
@@ -0,0 +1,334 @@
...
+public class MaterializedViewUtils {
+
+    /**
+     * Get related base table info which materialized view plan column reference,
+     * input param plan should be rewritten plan that sub query should be eliminated
+     *
+     * @param materializedViewPlan this should be rewritten or analyzed plan, should not be physical plan.
+     * @param column ref column name.
+     */
+    public static RelatedTableInfo getRelatedTableInfo(String column, Plan materializedViewPlan) {
```
Comment (zddr) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java:60]: The return value may be empty, is Optional more appropriate?

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+            this.relatedTableColumn = relatedTableColumn;
+        }
+    }
+
+    /**
+     * The related table info that mv relate
+     */
+    public static final class RelatedTableInfo {
+        private BaseTableInfo tableInfo;
+        private boolean pctPossible;
```
Comment (zddr) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java:313]: What does this mean

File: fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java (hunk 1 above)
Comment (seawinde) [reply fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/MaterializedViewUtils.java:60]: ok , i will fix it

**Comment by seawinde** [comment]
run buildall

**Review by zddr — APPROVED** [review]
LGTM

**Review by keanji-x — APPROVED** [review]

**Review by morrySnow — APPROVED** [review]

**Pull request merged**